packages/skill-api/src/core/services/create-magic-link.ts

```diff
@@ -0,0 +1,84 @@
+import {z} from 'zod'
+import {IncomingRequest, OutgoingResponse} from '../index'
+import {SkillRecordingsHandlerParams} from '../types'
+import {createVerificationUrl} from '../../server'
+
+class AuthorizationError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'ValidationError'
+  }
+}
+
+const requireSkillSecret = (req: IncomingRequest): void => {
+  try {
+    const verifySkillSecretHeader = z
+      .object({skillSecretInHeader: z.string(), skillSecret: z.string()})
+      .transform(({skillSecretInHeader, skillSecret}, ctx) => {
+        if (skillSecret !== skillSecretInHeader)
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: 'Unauthorized request to skill-api',
+          })
+      })
+
+    verifySkillSecretHeader.parse({
+      skillSecretInHeader: req.headers['x-skill-secret'],
+      skillSecret: process.env.SKILL_SECRET,
+    })
+  } catch (error: any) {
+    if (error instanceof z.ZodError) {
+      throw new AuthorizationError(error.message)
+    }
+
+    throw error
+  }
+}
```

Comment:
I modeled this skill-secret check after the ones already used for endpoints like `process-refund`. I pulled it out into its own function to make it a little more robust and tighten the types. I like where I got to with this and think it could become a shared function used across these other endpoints:

<img width="568" alt="CleanShot 2024-02-06 at 12 18 24@2x" src="https://github.com/skillrecordings/products/assets/694063/b0e5d80f-e8d3-4b3f-b6f8-e20056540d18">